README.md

```diff
@@ -17,3 +17,329 @@ Implement a prototype job worker service that provides an API to run arbitrary L
 
 * Client should be able to query result of the job execution and fetch the logs
 
+## Design Document
+
+### Limitations And Scope
+
+#### Data Management
+Though the use of a database to store persistant data would be ideal, I will be instead storing the outputs and error outputs into logs stored on 
+the file storage of the linux worker. The logs will be generated at start time with the foldername \<pid\>_\<startingTimestamp\>
+The client will have to maintain a queue of commands, the client can query the server to see what is running to determine what can be killed,
+or the client can store response data from server.
+
+#### Scale
+The scope of this project would only deal with a single linux worker server interfacing with one client
+
+### API Design
+
+* The client will have an Command Api that takes in three types of request, Start, Stop, and Query.
+
+* The client can then execute the commands over the server via Execute Api
+
+```
+type CommandRequest {
+    variableTypeRequest {
```

Comment:
what's a `variableTypeRequest` in gRPC terms? is it `oneof`?
if so, please use separate RPCs for start/stop/querypid/queryrunningprocesses instead

Reply:
will do